Expensify/App
# Add Expensify.cash to iOS and Android native share menus

### Details
Uses [react-native-share-menu](https://github.com/meedan/react-native-share-menu) to add Expensify.cash to the native share menu on mobile platforms and enables sharing to the app.

Chat selection is made on separate page, which reuses components from Home page. To achieve this I had to modify some sidebar related components and functions, mainly to add callbacks for getting selected chat id.

### Fixed Issues
Fixes #1294

### Tests
1. Open some other app on iOS or Android, for example image gallery or web browser.
2. Activate share menu for the element that needs to be shared.
3. Confirm that Expensify.cash is present in Share menu.
4. Select Expensify.cash.
5. Confirm that Expensify.cash is opened and routes to `Send to...` page.
6. Select or create and select some chat.
7. Confirm that shared item is posted to selected chat.

### Tested On


### Screenshots
#### iOS
https://user-images.githubusercontent.com/130532/106333690-e499d980-6291-11eb-9d99-68ea9473a482.mp4

#### Android
https://user-images.githubusercontent.com/130532/106398712-3327af00-641d-11eb-9106-fe0def954ad9.mp4
